lib/acts_as_taggable_on_mongoid/tag_list.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+# require "active_support/core_ext/module/delegation"
+
+module ActsAsTaggableOnMongoid
+  # A list of tags.  The TagList must be initialized with a tag definition so that it knows how to clean
+  # the list properly and to convert the list to a string.
+  #
+  # All methods that add objects to the list (initialization, concat, etc.) optionally take an array of values including
+  # options to parse the values and to optionally specifiy the parser to use.  If no parser is specified, then
+  # parser for the tag_definition is used.
+  #
+  # If the input value(s) are to be parsed, then all values passed in are parsed.
+  #
+  # Examples:
+  #   TagList.new(tag_definition, "value 1", "value 2")
+  #   # > TagList<> ["value 1", "value 2"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", parse: true)
+  #   # > TagList<> ["value 1", "value 2"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", "value 3, value 4", parse: true)
+  #   # > TagList<> ["value 1", "value 2", "value 3", "value 4"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", "value 3, value 4", parser: ActsAsTaggableOnMongoid::GenericParser)
+  #   # > TagList<> ["value 1", "value 2", "value 3", "value 4"]
+
+  # :reek:SubclassedFromCoreClass
+  class TagList < Array
+    # :reek:Attribute
+    attr_accessor :taggable
```

Comment:
I use this in Taggable objects so that the list can notify the object if it changes (and is therefore dirty.)

Based on what I was looking at in the ActsAsTaggable gem, this was handled by ActiveRecord fairly seamlessly.

It may be possible to get some similar "for free" functionality from Mongoid, but it would require other changes - changes which I chose not to do as I felt that they would be more potentially damaging/dangerous than this/the route I chose.